search/search_integration_tests/processor_test.cpp

```diff
@@ -1098,5 +1099,35 @@ UNIT_CLASS_TEST(ProcessorTest, Cian)
     TEST(ResultsMatch(params, rules), ());
   }
 }
+
+UNIT_CLASS_TEST(ProcessorTest, CityBoundaryLoad)
+{
+  TestCity city(vector<m2::PointD>({m2::PointD(0, 0), m2::PointD(0.5, 0), m2::PointD(0.5, 0.5),
+                                    m2::PointD(0, 0.5)}),
+                "moscow", "en", 100 /* rank */);
+  auto const id = BuildWorld([&](TestMwmBuilder & builder) { builder.Add(city); });
+
+  SetViewport(m2::RectD(m2::PointD(-1.0, -1.0), m2::PointD(1.0, 1.0)));
+  {
+    TRules const rules{ExactMatch(id, city)};
+    TEST(ResultsMatch("moscow", "en", rules), ());
+  }
+
+  CitiesBoundariesTable table;
+  TEST(table.Load(m_engine), ());
+  TEST(table.Has(0 /* fid */), ());
+  TEST(!table.Has(10 /* fid */), ());
+
+  CitiesBoundariesTable::Boundaries boundaries;
+  TEST(table.Get(0 /* fid */, boundaries), ());
+  TEST(boundaries.HasPoint(m2::PointD(0, 0)), ());
```

Comment:
Now here it looks unfortunate: a box has a point when the point is inside, but a boundary has measure zero.
I'm not sure what the good name would be, so please add a comment to Boundary's HasPoint that it checks that the point is inside the region bounded by |this|.

Reply:
I added the comment.